dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java

```diff
@@ -283,6 +306,15 @@ public String getFileName()
      */
     private void createDataParts() throws HpccFileException
     {
+        long fileAccessExpiryMS = fileAccessExpirySecs * 1000;
+        long dataPartsAgeMS = System.currentTimeMillis() - dataPartsCreationTimeMS;
+        if (dataParts != null && dataPartsAgeMS < fileAccessExpiryMS)
+        {
+            return;
+        }
```

Comment:
if datapars !=null, but dataPartsAgeMS  >= fileAccessExpiryMS, let's output a trace statement explaining why we're refreshing the hpccfile

Reply:
@rpastrana updated to log when refresh due to expiration